RespiratoryAssistDevices/R4/files/RespiratoryAssistDevicesFaceToFacePrepopulation-0.1.0.cql

```diff
@@ -0,0 +1,117 @@
+library RespiratoryAssistDevicesFaceToFacePrepopulation  version '0.1.0'
+using FHIR version '4.0.0'
+include FHIRHelpers version '4.0.0' called FHIRHelpers
+
+codesystem "LOINC": 'http://loinc.org'
+codesystem "HCPCS": 'https://bluebutton.cms.gov/resources/codesystem/hcpcs'
+
+
+//Respiratory Assist Device Qualifying Condition grouping value set
+valueset "RAD Qualifying Condition": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.122'
+
+//Motor Neuron Disease or Spinal Muscular Atrophy Disorder
+valueset "Neuromuscular Disease": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.44'
+//Chronic Obstructive Lung Disease (COPD)
+valueset "COPD": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.6'
+//Hypoventilation Disorder
+valueset "Hypoventilation Disorder": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.121'
+//Sleep Apnea or Breathing Related Sleep Disorder
+valueset "Apnea Sleep Disorder": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.31'
+
+//Apnea Hypopnea Index Rate Measurement
+valueset "AHI_Codes": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.123'
+//Maximal Inspiratory Pressure Measurement
+valueset "MIP_Codes": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.125'
+//Forced Vital Capacity (FVC) Measurement Prediction
+valueset "FVC_Codes": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.131'
+//Forced Expiratory Volume in One Second (FEV1)/Forced Vital Capacity (FVC) Measurement
+valueset "FEV1_FVC_Ratio_Codes": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.124'
+
+
+// LOINC codes for observations
+// Circumference Neck
+code "56074-8": '56074-8' from "LOINC"
+// Body mass index (BMI) [Ratio]
+code "39156-5": '39156-5' from "LOINC"
+
+parameter device_request DeviceRequest
+
+
+context Patient
+
+define Today: Today()
+
+
+define "EncounterDate":
```

Comment:
For EncounterDate, it should be the Encounter referred by the DeviceRequest if it exists.

Reply:
I was just following the comment "Find most recent encounter" given under the 'Notes' column in the 'DRLS Rule Sets - Primary DRAFT.xlsx' file. 
I will make the change like this: First find the Encounter referred by the DeviceRequest. If DeviceRequest does not point to any Encounter, find the most recent encounter. If no encounters exist, use Today.